apps/wfo-ui-surf/components/WfoServiceTicketsList/WfoServiceTicketsList.tsx

```diff
@@ -0,0 +1,253 @@
+    parseDateToLocaleDateTimeString,
+    parseIsoString,
+    SortOrder,
+    STATUS_COLOR_FIELD_COLUMN_PROPS,
+    TableColumnKeys,
+    useFilterQueryWithRest,
+    WfoBasicTable,
+    WfoDataSorting,
+    WfoDateTime,
+    WfoStatusColorField,
+    WfoTableColorColumnConfig,
+    WfoTableColumns,
+} from '@orchestrator-ui/orchestrator-ui-components';
+import { CIM_TICKETS_ENDPOINT } from '../../constants-surf';
+import { WfoServiceTicketStatusBadge } from '../WfoBadges/WfoServiceTicketStatusBadge';
+
+const SERVICE_TICKET_FIELD_JIRA_ID: keyof ServiceTicketDefinition =
+    'jira_ticket_id';
+const SERVICE_TICKET_FIELD_TITLE: keyof ServiceTicketDefinition = 'title_nl';
+const SERVICE_TICKET_FIELD_PROCESS_STATE: keyof ServiceTicketDefinition =
+    'process_state';
+const SERVICE_TICKET_FIELD_OPENED_BY: keyof ServiceTicketDefinition =
+    'opened_by';
+const SERVICE_TICKET_FIELD_START_DATE: keyof ServiceTicketDefinition =
+    'start_date';
+const SERVICE_TICKET_FIELD_CREATE_DATE: keyof ServiceTicketDefinition =
+    'create_date';
+const SERVICE_TICKET_FIELD_LAST_UPDATE: keyof ServiceTicketDefinition =
+    'last_update_time';
+
+const { NEW, OPEN, OPEN_RELATED, OPEN_ACCEPTED, UPDATED, ABORTED, CLOSED } =
+    ServiceTicketProcessState;
+
+export type WfoServiceTicketsListProps = {
+    alwaysOnFilters?: FilterQuery<ServiceTicketDefinition>[];
+    defaultHiddenColumns: TableColumnKeys<ServiceTicketDefinition> | undefined;
+    localStorageKey: string;
+    dataDisplayParams: DataDisplayParams<ServiceTicketDefinition>;
+    setDataDisplayParam: <
+        DisplayParamKey extends
+            keyof DataDisplayParams<ServiceTicketDefinition>,
+    >(
+        prop: DisplayParamKey,
+        value: DataDisplayParams<ServiceTicketDefinition>[DisplayParamKey],
+    ) => void;
+    overrideDefaultTableColumns?: (
+        defaultTableColumns: WfoTableColumns<ServiceTicketDefinition>,
+    ) => WfoTableColumns<ServiceTicketDefinition>;
+};
+
+export const WfoServiceTicketsList = ({
+    alwaysOnFilters,
+}: WfoServiceTicketsListProps) => {
+    const [pageIndex, setPageIndex] = useState(0);
+    const [pageSize, setPageSize] = useState(5);
+    const [dataSorting, setDataSorting] = useState<
+        WfoDataSorting<ServiceTicketDefinition>
+    >({
+        field: SERVICE_TICKET_FIELD_LAST_UPDATE,
+        sortOrder: SortOrder.DESC,
+    });
+
+    const t = useTranslations('cim.serviceTickets');
+    const { data, isFetching } = useFilterQueryWithRest(
+        CIM_TICKETS_ENDPOINT,
+        'serviceTickets',
+        alwaysOnFilters,
+    );
+
+    const serviceTicketColorMappings = {
+        success: [OPEN, NEW],
+        warning: [OPEN_ACCEPTED, OPEN_RELATED],
+        primary: [UPDATED],
+        lightShade: [CLOSED, ABORTED],
+    };
+
+    const tableColumns: WfoTableColumns<ServiceTicketDefinition> = {
+        status_color_field: {
+            ...(STATUS_COLOR_FIELD_COLUMN_PROPS as WfoTableColorColumnConfig<ServiceTicketDefinition>),
```

Comment:
Why would the type cast be needed here? By right the table component can be used without the need for type casting

Reply:
Fixed